db/migrate/20240712090133_add_index_to_fees.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class AddIndexToFees < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def change
+    unless index_exists?(:fees, [:charge_id, :invoice_id], name: 'index_fees_on_charge_id_and_invoice_id', where: 'deleted_at IS NULL')
+      add_index :fees, [:charge_id, :invoice_id],
+        name: 'index_fees_on_charge_id_and_invoice_id',
+        where: 'deleted_at IS NULL'
```

Comment:
I Think you can use `if_not_exists: true`
Also I think you missed the algorithm for the index creation